runtime/lua/vim/lsp/protocol.lua

```diff
@@ -318,6 +318,96 @@ for k, v in pairs(constants) do
   protocol[k] = tbl
 end
 
+--Lsp methods (based on version 3.18)
+protocol.Methods = {
```

Comment:
Can it be generated from `scripts/lsp_types.lua`?

Reply:
Out of curiosity @justinmk, have you guys considered using the [metamodel](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#metaModel) for LSP type generation? Not sure if it would make things easier in the Lua side, but at least it would help ensuring that Neovim's implementation remains up to date.

Reply:
Nice, didn't know about that! `metaModel.json` looks useful

Reply:
these methods generate from 3.18 modeljson https://microsoft.github.io/language-server-protocol/specifications/lsp/3.18/metaModel/metaModel.json

Reply:
forget `protococol.lua` is generate from script. yes we can also generate these methods from `lsp_types.lua` do it later.

Reply:
yeah it looks like `lsp.json` which is used by `lsp_types.lua`, is equivalent to `metaModel.json`

Reply:
I think we got one thing wrong. The file that `lsp_types` generates annotations is `lsp/types/protocol.lua`, and the constant file we need is` lsp/protocol.lua`. So I made some changes here in the script. For the previous script annotation generation code, I feel that it may need to be improved. compare with my implementation.